python/cudf/pyproject.toml

```
@@ -28,7 +28,7 @@ requires-python = ">=3.9"
 dependencies = [
     "cachetools",
     "cubinlinker",
-    "cuda-python>=11.7.1,<12.0",
+    "cuda-python>=11.7.1,<12.0a",
```

Looking at [RMM's `cuda-python` constraint]( https://github.com/rapidsai/rmm/blob/d6cc9b8599caaea9beb5aad1eaabe80cd363a91d/python/pyproject.toml#L19 ), think we want this

```suggestion
    "cuda-python>=11.7.1,<12.0a0",
```

Do we need to do anything special here to get a different `cuda-python` for CUDA 12?

---

`12.0a` is a typo on my part. I'll fix that.

For `pyproject.toml`, currently this is handled by https://github.com/rapidsai/cudf/blob/branch-23.08/ci/release/apply_wheel_modifications.sh until `rapids-dependency-file-generator` supports `pyproject.toml` with matrices.

---

Typo fixed in 30d018cf.